include/mqtt/server.hpp

```diff
@@ -84,9 +84,25 @@ class server {
     void listen() {
         close_request_ = false;
         renew_socket();
+
+        auto post_error =
+            [&](boost::system::error_code const& ec) {
```

Comment:
Why the lambda? It's only used in one place (inside the try-catch).

It would be simpler to call ios_accept_.post() directly, wouldn't it?

Reply:
@jonesmz I will fix it as you pointed out.

I tried open, set_option, bind, listen, and then accept approach before, but didn't work as I expected. The lambda was used multiple times at that approach. I forgot to update it.